spring-xd-rest-client/src/main/java/org/springframework/xd/rest/client/impl/StreamTemplate.java

```diff
@@ -95,4 +97,17 @@ public void destroyAll() {
 		restTemplate.delete(resources.get("streams/definitions").expand());
 	}
 
+	@Override
+	public void deleteQueue(String name, Map<String, String> properties) {
+		String rootUri = resources.get("streams/clean/rabbit").toString() + "/{name}";
+		UriComponentsBuilder builder = UriComponentsBuilder.fromUriString(rootUri);
+
+		for(Map.Entry<String, String> entry: properties.entrySet()) {
+			builder.queryParam(entry.getKey(), entry.getValue());
+		}
```

Comment:
It would be nice if the template could have some (maybe) all of these properties wired in as properties since they are likely to be the same for each call.

Ditto for the job implementation - however, it would be better to pull the guts of this method up to `AbstractTemplate` since the only difference between the two templates is the `rootUri`.